platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -1150,6 +1161,18 @@ - (id)mgl_jsonExpressionObject {
                 } else if ([type isEqualToString:@"NSNumber"]) {
                     return @[@"to-number", object];
                 }
+#if TARGET_OS_IPHONE
+                else if ([type isEqualToString:@"UIColor"] || [type isEqualToString:@"MGLColor"]) {
+                    return @[@"to-color", object];
+                }
+#else
+                else if ([type isEqualToString:@"NSColor"] || [type isEqualToString:@"MGLColor"]) {
+                    return @[@"to-color", object];
+                }
+#endif
+                else if ([type isEqualToString:@"NSArray"]) {
+                    return @[@"to-rgba", object];
```

Comment:
`to-rgba` is only appropriate if the object is a color. If it’s a constant UIColor or NSColor, we can use `to-rgba` for sure. If it’s a different kind of constant value, such as an NSString, then we should raise an exception since that kind of conversion is currently unsupported. For non-constant expressions, we don’t know the resolved type for sure, so we can use `to-rgba` in that case.